src/main/java/com/palantir/docker/compose/execution/SynchronousDockerComposeExecutable.java

```diff
@@ -15,61 +15,13 @@
  */
 package com.palantir.docker.compose.execution;
 
-import static com.google.common.base.Throwables.propagate;
-import static java.nio.charset.StandardCharsets.UTF_8;
-import static java.util.concurrent.Executors.newSingleThreadExecutor;
-import static java.util.stream.Collectors.joining;
-
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Future;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
 import java.util.function.Consumer;
 
-public class SynchronousDockerComposeExecutable {
-    public static final int HOURS_TO_WAIT_FOR_STD_OUT_TO_CLOSE = 12;
-    public static final int MINUTES_TO_WAIT_AFTER_STD_OUT_CLOSES = 1;
-    private final DockerComposeExecutable dockerComposeExecutable;
-    private final Consumer<String> logConsumer;
+public class SynchronousDockerComposeExecutable extends AbstractSynchronousExecutable<DockerComposeExecutable> {
```

Comment:
With the changes I suggested above to remove generics I think this class is no longer needed which makes things a lot simpler and easier for somebody new to the codebase.